runtime/src/lib.rs

```diff
@@ -74,6 +74,15 @@ impl From<RtBootStatus> for u32 {
 
 pub const DPE_SUPPORT: Support = Support::all();
 
+pub struct CertChain {
+    pub cert: [u8; Self::MAX_CERT_CHAIN_SIZE],
+    pub cert_chain_size: usize,
+}
+
+impl CertChain {
+    const MAX_CERT_CHAIN_SIZE: usize = 4096;
+}
```

Comment:
Maybe ArrayVec would be better here. Then you can use `try_push` to push bytes into it.

Reply:
Done. However it did result in extra stack allocation since I couldn't figure out a way to safely pass the ArrayVec to copy_fmc_alias_cert, copy_ldevid_cert, etc.